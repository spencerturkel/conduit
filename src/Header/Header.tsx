import * as React from 'react';

const Header = () => (
    <nav className="navbar navbar-light">
        <div className="container">
            <a className="navbar-brand" href="/">
                conduit
            </a>
            <ul className="nav navbar-nav pull-xs-right">
                <li className="nav-item">
                    {/* TODO: Set active based on route */}
                    <a className="nav-link active" href="">
                        Home
                    </a>
                </li>
                <li className="nav-item">
                    <a className="nav-link" href="">
                        <i className="ion-compose" />&nbsp;New Post
                    </a>
                </li>
                <li className="nav-item">
                    <a className="nav-link" href="">
                        <i className="ion-gear-a" />&nbsp;Settings
                    </a>
                </li>
                <li className="nav-item">
                    <a className="nav-link" href="">
                        Sign up
                    </a>
                </li>
            </ul>
        </div>
    </nav>
);

export default Header;
