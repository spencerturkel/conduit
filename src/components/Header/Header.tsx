import * as React from 'react';
import {ComponentType, StatelessComponent} from 'react';

import {NavLinkProps} from '../nav-link-props';
import setDisplayName from '../set-display-name';

const Header = (Link: ComponentType<NavLinkProps>): StatelessComponent => {
    const sfc: StatelessComponent = () => (
        <nav className="navbar navbar-light">
            <div className="container">
                <Link className="navbar-brand" to="/">
                    conduit
                </Link>
                <ul className="nav navbar-nav pull-xs-right">
                    <li className="nav-item">
                        <Link activeClassName="active" className="nav-link" to="/">
                            Home
                        </Link>
                    </li>
                    <li className="nav-item">
                        <Link activeClassName="active" className="nav-link" to="/new-post">
                            <i className="ion-compose" />&nbsp;New Post
                        </Link>
                    </li>
                    <li className="nav-item">
                        <Link activeClassName="active" className="nav-link" to="/settings">
                            <i className="ion-gear-a" />&nbsp;Settings
                        </Link>
                    </li>
                    <li className="nav-item">
                        <Link activeClassName="active" className="nav-link" to="/sign-up">
                            Sign up
                        </Link>
                    </li>
                </ul>
            </div>
        </nav>
    );

    setDisplayName(sfc, Header, Link);

    return sfc;
};

export default Header;
