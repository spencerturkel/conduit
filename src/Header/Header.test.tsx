import {shallow} from 'enzyme';
import * as React from 'react';

import Header from './Header';

it('matches snapshot', () => {
    const sut = shallow(<Header />);
    expect(sut).toMatchSnapshot();
});
