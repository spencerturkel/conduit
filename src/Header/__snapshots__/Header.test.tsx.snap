// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Header />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="container"
>
        <a
                className="navbar-brand"
                href="/"
        >
                conduit
        </a>
        <ul
                className="nav navbar-nav pull-xs-right"
        >
                <li
                        className="nav-item"
                >
                        <a
                                className="nav-link active"
                                href=""
                        >
                                Home
                        </a>
                </li>
                <li
                        className="nav-item"
                >
                        <a
                                className="nav-link"
                                href=""
                        >
                                <i
                                        className="ion-compose"
                                />
                                 New Post
                        </a>
                </li>
                <li
                        className="nav-item"
                >
                        <a
                                className="nav-link"
                                href=""
                        >
                                <i
                                        className="ion-gear-a"
                                />
                                 Settings
                        </a>
                </li>
                <li
                        className="nav-item"
                >
                        <a
                                className="nav-link"
                                href=""
                        >
                                Sign up
                        </a>
                </li>
        </ul>
</div>,
      "className": "navbar navbar-light",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <a
            className="navbar-brand"
            href="/"
>
            conduit
</a>,
          <ul
            className="nav navbar-nav pull-xs-right"
>
            <li
                        className="nav-item"
            >
                        <a
                                    className="nav-link active"
                                    href=""
                        >
                                    Home
                        </a>
            </li>
            <li
                        className="nav-item"
            >
                        <a
                                    className="nav-link"
                                    href=""
                        >
                                    <i
                                                className="ion-compose"
                                    />
                                     New Post
                        </a>
            </li>
            <li
                        className="nav-item"
            >
                        <a
                                    className="nav-link"
                                    href=""
                        >
                                    <i
                                                className="ion-gear-a"
                                    />
                                     Settings
                        </a>
            </li>
            <li
                        className="nav-item"
            >
                        <a
                                    className="nav-link"
                                    href=""
                        >
                                    Sign up
                        </a>
            </li>
</ul>,
        ],
        "className": "container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "conduit",
            "className": "navbar-brand",
            "href": "/",
          },
          "ref": null,
          "rendered": "conduit",
          "type": "a",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <li
                className="nav-item"
>
                <a
                                className="nav-link active"
                                href=""
                >
                                Home
                </a>
</li>,
              <li
                className="nav-item"
>
                <a
                                className="nav-link"
                                href=""
                >
                                <i
                                                className="ion-compose"
                                />
                                 New Post
                </a>
</li>,
              <li
                className="nav-item"
>
                <a
                                className="nav-link"
                                href=""
                >
                                <i
                                                className="ion-gear-a"
                                />
                                 Settings
                </a>
</li>,
              <li
                className="nav-item"
>
                <a
                                className="nav-link"
                                href=""
                >
                                Sign up
                </a>
</li>,
            ],
            "className": "nav navbar-nav pull-xs-right",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <a
                  className="nav-link active"
                  href=""
>
                  Home
</a>,
                "className": "nav-item",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Home",
                  "className": "nav-link active",
                  "href": "",
                },
                "ref": null,
                "rendered": "Home",
                "type": "a",
              },
              "type": "li",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <a
                  className="nav-link"
                  href=""
>
                  <i
                                    className="ion-compose"
                  />
                   New Post
</a>,
                "className": "nav-item",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <i
                      className="ion-compose"
/>,
                    " New Post",
                  ],
                  "className": "nav-link",
                  "href": "",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "className": "ion-compose",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": "i",
                  },
                  " New Post",
                ],
                "type": "a",
              },
              "type": "li",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <a
                  className="nav-link"
                  href=""
>
                  <i
                                    className="ion-gear-a"
                  />
                   Settings
</a>,
                "className": "nav-item",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <i
                      className="ion-gear-a"
/>,
                    " Settings",
                  ],
                  "className": "nav-link",
                  "href": "",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "className": "ion-gear-a",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": "i",
                  },
                  " Settings",
                ],
                "type": "a",
              },
              "type": "li",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <a
                  className="nav-link"
                  href=""
>
                  Sign up
</a>,
                "className": "nav-item",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Sign up",
                  "className": "nav-link",
                  "href": "",
                },
                "ref": null,
                "rendered": "Sign up",
                "type": "a",
              },
              "type": "li",
            },
          ],
          "type": "ul",
        },
      ],
      "type": "div",
    },
    "type": "nav",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="container"
>
          <a
                    className="navbar-brand"
                    href="/"
          >
                    conduit
          </a>
          <ul
                    className="nav navbar-nav pull-xs-right"
          >
                    <li
                              className="nav-item"
                    >
                              <a
                                        className="nav-link active"
                                        href=""
                              >
                                        Home
                              </a>
                    </li>
                    <li
                              className="nav-item"
                    >
                              <a
                                        className="nav-link"
                                        href=""
                              >
                                        <i
                                                  className="ion-compose"
                                        />
                                         New Post
                              </a>
                    </li>
                    <li
                              className="nav-item"
                    >
                              <a
                                        className="nav-link"
                                        href=""
                              >
                                        <i
                                                  className="ion-gear-a"
                                        />
                                         Settings
                              </a>
                    </li>
                    <li
                              className="nav-item"
                    >
                              <a
                                        className="nav-link"
                                        href=""
                              >
                                        Sign up
                              </a>
                    </li>
          </ul>
</div>,
        "className": "navbar navbar-light",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <a
              className="navbar-brand"
              href="/"
>
              conduit
</a>,
            <ul
              className="nav navbar-nav pull-xs-right"
>
              <li
                            className="nav-item"
              >
                            <a
                                          className="nav-link active"
                                          href=""
                            >
                                          Home
                            </a>
              </li>
              <li
                            className="nav-item"
              >
                            <a
                                          className="nav-link"
                                          href=""
                            >
                                          <i
                                                        className="ion-compose"
                                          />
                                           New Post
                            </a>
              </li>
              <li
                            className="nav-item"
              >
                            <a
                                          className="nav-link"
                                          href=""
                            >
                                          <i
                                                        className="ion-gear-a"
                                          />
                                           Settings
                            </a>
              </li>
              <li
                            className="nav-item"
              >
                            <a
                                          className="nav-link"
                                          href=""
                            >
                                          Sign up
                            </a>
              </li>
</ul>,
          ],
          "className": "container",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "conduit",
              "className": "navbar-brand",
              "href": "/",
            },
            "ref": null,
            "rendered": "conduit",
            "type": "a",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <li
                  className="nav-item"
>
                  <a
                                    className="nav-link active"
                                    href=""
                  >
                                    Home
                  </a>
</li>,
                <li
                  className="nav-item"
>
                  <a
                                    className="nav-link"
                                    href=""
                  >
                                    <i
                                                      className="ion-compose"
                                    />
                                     New Post
                  </a>
</li>,
                <li
                  className="nav-item"
>
                  <a
                                    className="nav-link"
                                    href=""
                  >
                                    <i
                                                      className="ion-gear-a"
                                    />
                                     Settings
                  </a>
</li>,
                <li
                  className="nav-item"
>
                  <a
                                    className="nav-link"
                                    href=""
                  >
                                    Sign up
                  </a>
</li>,
              ],
              "className": "nav navbar-nav pull-xs-right",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <a
                    className="nav-link active"
                    href=""
>
                    Home
</a>,
                  "className": "nav-item",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Home",
                    "className": "nav-link active",
                    "href": "",
                  },
                  "ref": null,
                  "rendered": "Home",
                  "type": "a",
                },
                "type": "li",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <a
                    className="nav-link"
                    href=""
>
                    <i
                                        className="ion-compose"
                    />
                     New Post
</a>,
                  "className": "nav-item",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      <i
                        className="ion-compose"
/>,
                      " New Post",
                    ],
                    "className": "nav-link",
                    "href": "",
                  },
                  "ref": null,
                  "rendered": Array [
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "className": "ion-compose",
                      },
                      "ref": null,
                      "rendered": null,
                      "type": "i",
                    },
                    " New Post",
                  ],
                  "type": "a",
                },
                "type": "li",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <a
                    className="nav-link"
                    href=""
>
                    <i
                                        className="ion-gear-a"
                    />
                     Settings
</a>,
                  "className": "nav-item",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      <i
                        className="ion-gear-a"
/>,
                      " Settings",
                    ],
                    "className": "nav-link",
                    "href": "",
                  },
                  "ref": null,
                  "rendered": Array [
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "className": "ion-gear-a",
                      },
                      "ref": null,
                      "rendered": null,
                      "type": "i",
                    },
                    " Settings",
                  ],
                  "type": "a",
                },
                "type": "li",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <a
                    className="nav-link"
                    href=""
>
                    Sign up
</a>,
                  "className": "nav-item",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Sign up",
                    "className": "nav-link",
                    "href": "",
                  },
                  "ref": null,
                  "rendered": "Sign up",
                  "type": "a",
                },
                "type": "li",
              },
            ],
            "type": "ul",
          },
        ],
        "type": "div",
      },
      "type": "nav",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
